apiserver/facades/client/modelmanager/register.go

```diff
@@ -86,6 +90,16 @@ func newFacadeV10(stdCtx context.Context, ctx facade.MultiModelContext) (*ModelM
 	backend := common.NewUserAwareModelManagerBackend(configSchemaSource, model, pool, apiUser)
 
 	secretBackendService := domainServices.SecretBackend()
+
+	bootstrapParamsData, err := os.ReadFile(path.Join(ctx.DataDir(), cloudconfig.FileNameBootstrapParams))
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
```

Comment:
Do we need to do this file reading exactly here? at the facades' api constructor? 
I'm guessing the contents of this file won't change once the apiserver is started, so could we do it in the worker only once?

Reply:
We can't do this here, this won't work in HA as the other nodes won't have this file. You need to store all the information you need in database at the bootstrap worker stage and then read it in from here.

Reply:
@SimonRichardson Oh right, i wasn't aware of that (being naïve ^^)

For me there is two solution here:

* Retrieve the custom image from Controller model (which means that we need to have a CloudImageMetadataService linked to this other model db here), which is provisionned in the bootstrap worker. It will be the least impactful amho. It will came with the side-effect that adding a custom image meta data into the controller model through the metadata plugin will spawn it in all new model created after. It may or may not be desired (but doesn't seems harmful at first sight).
* Duplicating the db in the controller, having it provisionned during bootstrap and use the controller db to provision all model db. It will be more impactful (it basically duplicates the schema, and so the domain/state test, and requires another NewService function which target the controller.

I am personally inclined to the former.

Reply:
Removed the last commit which introduces this issue and replace the logic with loading metadata from controller model db. The controller model db was loaded before through bootstrap logic.

Reply:
Out of sync => We will load metadata in controller db instead of controller model db.